### PR TITLE
Implement common traits for `Layer`

### PR DESCRIPTION
**Pull request by madsmtm**
Implement `Debug`, `Clone`, `PartialEq`, `Eq`, `Hash`, `Send`, `Sync`, `UnwindSafe` and `RefUnwindSafe` for `Layer`.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -14,11 +16,43 @@ pub mod appkit;
...
+#[derive(Debug, Clone)]
 pub struct Layer {
     layer: Retained<CAMetalLayer>,
     pre_existing: bool,
 }
 
+impl PartialEq for Layer {
+    #[inline]
+    fn eq(&self, other: &Self) -> bool {
+        self.layer == other.layer
```
Comment (MarijnS95) [inline src/lib.rs]: Is `.eq()` clearer?

File: src/lib.rs [hunk 2]
```diff
@@ -3,9 +3,11 @@
 #![cfg_attr(docsrs, feature(doc_auto_cfg, doc_cfg_hide), doc(cfg_hide(doc)))]
 #![deny(unsafe_op_in_unsafe_fn)]
 
+use core::ffi::c_void;
+use core::hash;
+use core::panic::{RefUnwindSafe, UnwindSafe};
 use objc2::rc::Retained;
 use objc2_quartz_core::CAMetalLayer;
-use std::ffi::c_void;
```
Comment (MarijnS95) [inline src/lib.rs:8]: Can we `#![no_std]`?

**Review by MarijnS95 — APPROVED** [review]

File: src/lib.rs (hunk 2 above)
Comment (madsmtm) [reply src/lib.rs:8]: Not really yet, `objc2` isn't `#![no_std]`.

File: src/lib.rs (hunk 1 above)
Comment (madsmtm) [reply src/lib.rs]: Sure

**Pull request merged**